compose/legacy.py

```diff
@@ -24,41 +25,81 @@
     $ docker rm -f {rm_args}
 """
 
+ONE_OFF_ADDENDUM_FORMAT = """
+You should also remove your one-off containers:
+
+    $ docker rm -f {rm_args}
+"""
+
+ONE_OFF_ERROR_MESSAGE_FORMAT = """
+Compose found the following containers without labels:
+
+{names_list}
+
+As of Compose 1.3.0, containers are identified with labels instead of naming convention.
+
+Remove them before continuing:
+
+    $ docker rm -f {rm_args}
+"""
+
 
 def check_for_legacy_containers(
         client,
         project,
         services,
-        stopped=False,
-        one_off=False):
+        allow_one_off=True):
     """Check if there are containers named using the old naming convention
     and warn the user that those containers may need to be migrated to
     using labels, so that compose can find them.
     """
-    containers = list(get_legacy_containers(
+    (containers, one_off_containers) = get_legacy_containers(
         client,
         project,
         services,
-        stopped=stopped,
-        one_off=one_off))
+    )
 
     if containers:
-        raise LegacyContainersError([c.name for c in containers])
+        raise LegacyContainersError(
+            [c.name for c in containers],
+            [c.name for c in one_off_containers],
+        )
+
+    if one_off_containers and not allow_one_off:
+        raise LegacyOneOffContainersError(
+            [c.name for c in one_off_containers],
+        )
```

Comment:
With the suggested edit to `get_legacy_containers`, I think this would be

```python

containers = get_legacy_containers(..., one_off=False)
if containers:
    raise ...

if not allow_one_off and get_legacy_containers(..., one_off=True):
    riase ...
```